elephant/spike_train_dissimilarity.py

```diff
@@ -0,0 +1,406 @@
+import elephant.kernels as kernels
+import cmath
+from neo.core import SpikeTrain
+
+# Problem of conversion from Python 2 to Python 3:
+# 'xrange' in Python 2 is 'range' in Python 3. 
+try:
+    xrange
+except NameError:
+    xrange = range
+
+
+def _create_matrix_from_indexed_function(
+        shape, func, symmetric_2d=False, **func_params):
+    mat = np.empty(shape)
+    if symmetric_2d:
+        for i in xrange(shape[0]):
+            for j in xrange(i, shape[1]):
+                mat[i, j] = mat[j, i] = func(i, j, **func_params)
+    else:
+        for idx in np.ndindex(*shape):
+            mat[idx] = func(*idx, **func_params)
+    return mat
+
+
+def victor_purpura_dist(
+        trains, q=1.0 * pq.Hz, kernel=None, sort=True, algor='fast_algor'):
+    """ 
+    Calculates the Victor-Purpura's (VP) distance. It is often denoted as
+    :math:`D^{\\text{spike}}[q]`.
+
+    It is defined as the minimal cost of transforming spike train `a` into
+    spike train `b` by using the following operations:
+
+        * Inserting or deleting a spike (cost 1.0).
+        * Shifting a spike from :math:`t` to :math:`t'` (cost :math:`q 
+          \\cdot |t - t'|`).
+
+    A detailed description can be found in
+    *Victor, J. D., & Purpura, K. P. (1996). Nature and precision of
+    temporal coding in visual cortex: a metric-space analysis. Journal of
+    Neurophysiology.*
+
+    Given the average number of spikes :math:`n` in a spike train and
+    :math:`N` spike trains the run-time complexity of this function is
+    :math:`O(N^2 n^2)` and :math:`O(N^2 + n^2)` memory will be needed.
+
+    Parameters
+    ----------
+    trains : Sequence of :class:`neo.core.SpikeTrain` objects of
+        which the distance will be calculated pairwise.
+    q: Quantity scalar 
+        Cost factor for spike shifts as inverse time scalar. If `kernel`
+        is not `None`, :math:`q` will be ignored.
+        Default: 1.0 * pq.Hz
+    kernel: :class:`.kernels.Kernel`
+        Kernel to use in the calculation of the distance. If `kernel` is
+        `None`, an unnormalized triangular kernel with standard deviation
+        of :math:'2.0/(q * sqrt(6.0))' corresponding to a half width of 
+        :math:`2.0/q` will be used. Usage of the default value calculates
+        the Victor-Purpura distance correctly with a triangular kernel of
+        the suitable width. The choice of another kernel is enabled, but
+        this leaves the framework of Victor-Purpura distances.
+        Default: None
+    sort: bool 
+        Spike trains with sorted spike times will be needed for the
+        calculation. You can set `sort` to `False` if you know that your
+        spike trains are already sorted to decrease calculation time.
+        Default: True
+    algor: string
+        Allowed values are 'fast_algor' or 'intuitive_algor', each selecting an
+        algorithm with which to calculate the pairwise Victor-Purpura distance.
+        Default: 'fast_algor'
```

Comment:
I think the user should have a hint here, on what to choose. Although you argue down below why both algorithms are kept, this doc string should probably contain a small hint that for all practical purposes, fast_algor should be used, and that there is now problem in using the efficient version (i.e., identical results).

Also, I would suggest to use 'fast" and 'intutitive" instead of 'fast_algor' and 'intutitive_algor', since otherwise code will look double

...algor="fast_algor"

I would also suggest to use 'algorithm' as variable name, 'algor' seems a bit strange as an abbreviation (algor, algoris: coldness, frost).